include/general-redaction/rapidjsonRedactor.hpp

```diff
@@ -0,0 +1,138 @@
+#include <string>
+#include <iostream>
+#include "rapidjson/document.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+
+static const char* kTypeNames[] = { "Null", "False", "True", "Object", "Array", "String", "Number" };
+
+/**
+ * @brief A tool for redacting members of rapidjson::Document and rapidjson::Value objects
+ * 
+ */
+class RapidjsonRedactor {
+    public:
+        RapidjsonRedactor();
+
+        // redaction methods
+        
+        /**
+         * @brief Redacts all instances of a member by name
+         * 
+         * @param value The rapidjson::Value to redact from
+         * @param member The name of the member to redact
+         * @param success Whether or not the redaction was successful
+         */
+        void redactAllInstancesOfMemberByName(rapidjson::Value& value, std::string member, bool& success);
+
+        /**
+         * @brief Redacts all instances of a member by path
+         * 
+         * @param value The rapidjson::Value to redact from
+         * @param path The path to the member to redact
+         * @param success Whether or not the redaction was successful
+         */
+        void redactMemberByPath(rapidjson::Value& value, std::string path, bool& success);
+
+        /**
+         * @brief Searches for a member by name
+         * 
+         * @param value The rapidjson::Value to search
+         * @param member The name of the member to search for
+         * @param success Whether or not the search was successful
+         */
+        void searchForMemberByName(rapidjson::Value& value, std::string member, bool& success);
+
+        /**
+         * @brief Searches for a member by path
+         * 
+         * @param value The rapidjson::Value to search
+         * @param path The path to the member to search for
+         * @param success Whether or not the search was successful
+         */
+        void searchForMemberByPath(rapidjson::Value& value, std::string path, bool& success);
+
+        // utility methods
```

Comment:
Most of these utility methods don't seem to be used, should we remove them?

Reply:
The unused utility methods have been removed and the class has been simplified.